test/nodes/test_context_optimizer.py

```
@@ -0,0 +1,968 @@
+import pytest
+
+# ---------------------------------------------------------------------------
+# Stub external dependencies so the optimizer module can be imported without
+# a running RocketRide server or tiktoken installed in the test env.
+# ---------------------------------------------------------------------------
+
+_STUBS_INSTALLED = False
+
+
+def _install_stubs() -> None:
+    global _STUBS_INSTALLED
```

Ruff (0.15.7)

[warning] Using the global statement to update `_STUBS_INSTALLED` is discouraged

(PLW0603)